tests/integration/targets/version_update/tasks/main.yml

```diff
@@ -0,0 +1,52 @@
+---
+- environment:
+    SC_HOST: "{{ sc_host }}"
+    SC_USERNAME: "{{ sc_username }}"
+    SC_PASSWORD: "{{ sc_password }}"
+    SC_TIMEOUT: "{{ sc_timeout }}"
+  vars:
+    cluster_update_config: "{{ sc_config[sc_host].cluster_update }}"
+
+  block:
+    - name: Fail test early if update is not allowed
+      ansible.builtin.assert:
+        that:
+          - cluster_update_config.magic_allow_string == "allow-cluster-update-test"
+    
+    - name: Get list of hc version updates
+      scale_computing.hypercore.version_update_info:
+      register: updates
+    
+    - name: Get initial version
+      scale_computing.hypercore.cluster_info:
+      register: cluster_initial
+    
+    - name: Update hc version
+      scale_computing.hypercore.version_update:
+        icos_version: "{{ cluster_initial.record.icos_version}}"
+      register: update
+    - ansible.builtin.assert:
+        that:
+          - update is not changed
+          - not update.record
+          - update.diff.before.icos_version == cluster_initial.record.icos_version
+          - update.diff.after.icos_version == cluster_initial.record.icos_version
+
+    - name: Update hc version
+      scale_computing.hypercore.version_update:
+        icos_version: "{{ updates.records[0].uuid}}"
+      register: update
+    - ansible.builtin.assert:
+        that:
+          - update is changed
+          - update.record.uuid == updates.records[0].uuid
+          - update.diff.before.icos_version == cluster_initial.record.icos_version
+          - update.diff.after.icos_version == updates.records[0].uuid
+    
+    # - name: Get status of the latest update applied
```

Comment:
Now you check update is being applied? And do not expect it is already complete?

Reply:
Here I only confirm that the update was applied (if module exited without error), with module version_update_status_info we then check for the status (in progress, terminated, complete..)

Reply:
Added check with version_update_status_info if update is "in progress"

Reply:
@justinc1 I deleted this check, since it takes some time for the status to change and we don't really know how much time we have to wait - if we wait too little, than update_status will still be "COMPLETE" or "TERMINATING", if we wait too long then it can happen we skip the first change in the status (for example prepare_status="DOWNLOAD RPMS") and the update_status is then already "EXECUTING". Or it can even be "TERMINATING" if something went wrong early on. I think this check should only be implemented in a role.